### DISCUSSION TITLE
Weird issue with typed properties

### DISCUSSION THREAD

**Discussion by mokhosh** [livewire/livewire#3935] [7. Bugs (archived)] [answered]
I have a Livewire component that lets users request to be verified and get a blue tick on our platform. This is basically what the component looks like:

```php
class GetVerified extends Component
{
    public VerificationRequest $verificationRequest;

    protected $rules = [
        'verificationRequest.verification_type_id' => '',
    ];

    public function mount()
    {
        if (auth()->user()->isVerified())
            return redirect()->route('profile.show');

        $this->verificationRequest = auth()->user()->verification_request_draft();
        $this->verificationRequest ??= new VerificationRequest;
    }

    public function render()
    {
        return view('livewire.get-verified', [
            'documents' => $this->verificationRequest->fresh()?->getMedia() ?? [],
        ]);
    }
}
```

The intended behaviour is if the user is already verified the component is not rendered, instead the user is redirected to their profile, if not we show the form.
This works perfectly fine on local and on staging server, but it weirdly doesn't work on production server and says "$verificationRequest cannot be accessed before initialization".

I've tried these:
- Not use types: gives null error which isn't that much better
- change production environment to local, which didn't help
- update all servers to latest composer updates
- move `$this->verificationRequest ??= new VerificationRequest;` to render function AND not use types which helped, but is the weird part

**Comment by ousid** [comment]
Hey, @mokhosh 

Maybe try to make the `verificationRequest` protected instead of `public` since this property works only on the component!

```php
protected VerificationRequest $verificationRequest;
....
```

__PS__: I suggest moving this logic to normal controller instead of livewire component, since this component has no reaction with the frontend, which is `livewire` doesn't help in this case.

**Reply by mokhosh** [reply]
The component is a form and has a lot of interactive, i just copied the relevant parts to make it simpler to understand.

verificationRequest is edited through the form so it has to be public.

Thanks for your answer anyway.

**Comment by jskorlol** [comment]
This problem is also difficult for me to solve.

Before 2.6.0, it redirects without any problem, but after that, render proceeds unconditionally.

In case of redirect, ```$this->verificationRequest```, which is a subsequent operation, is not assigned.
However, because the render is executed unconditionally, the problem that there is no ```$this->verificationRequest``` appears.
It causes a terrible problem in situations where redirection is forced in multiple events. How can I solve this?

Livewire Controller

```php
    public function mount()
    {
        $this->items = \App\Models\Order::where('state', 2)->first()->toArray();
    }

    public function event1()
    {
        unset($this->items);
        return $this->redirectRoute('sale');
    }

    public function render()
    {
        return view('livewire.test');
    }
```

```html
        <div>
            <button wire:click="event1">Button</button>
            <div class="flex flex-col mt-4">
                model Binding
                <input type="text" wire:model="items.id">
            </div>
            <div class="flex flex-col mt-4">
                using
                <span> {{ $items['id'] }}</span>
            </div>
        </div>
```

```
ErrorException
Trying to access array offset on value of type null (View: /var/www/resources/views/livewire/test.blade.php)
```

This issue does not occur in 2.5.5.
But in 2.6.7 it happens...

When redirecting, I want to ignore render, is there any way?

In some event, there are cases where several model binding data are altered and saved, or redirected immediately after deleting the data..
Problems arise in this case...

Josh Hanley's solution

```$this->skipRender();```
It says to use .
```php
         if (auth()->user()->isVerified()) {
             $this->skipRender();
             return redirect()->route('profile.show');
             }
```
This way, when you need to redirect, it doesn't render.
Would you like to try it?

**Comment by mokhosh** [comment]
Thanks @jskorlol, I think that technically answers the question. I would love to get a bit of clarity about the internals and why this breaking change has been introduced into Livewire however. @joshhanley @calebporzio care to share?

**Comment by joshhanley** [comment]
@mokhosh yeah we changed how the redirect works due to issues with browser cache, but I hadn't anticipated the scenario you have and as such has introduced a breaking change. I'm working on a fix for it.

Hope this helps!

**Comment by joshhanley** [comment, marked as answer]
@mokhosh and @jskorlol I have submitted a PR #3971 with a fix for this.

Hope this helps!

**Comment by mokhosh** [comment]
Ok good to know. Thanks @joshhanley for all the amazing work.

**Reply by joshhanley** [reply]
No worries 🙂  PR has been merged!

**Reply by mokhosh** [reply]
Thanks, just waiting for a release then.
